src/components/AssistantPlainTextModal.vue

```diff
@@ -0,0 +1,258 @@
+import NcNoteCard from '@nextcloud/vue/dist/Components/NcNoteCard.js'
+
+import VueClipboard from 'vue-clipboard2'
+import Vue from 'vue'
+
+import { showError } from '@nextcloud/dialogs'
+
+import { TASK_TYPES } from '../constants.js'
+
+Vue.use(VueClipboard)
+
+export default {
+	name: 'AssistantPlainTextModal',
+	components: {
+		NcModal,
+		NcButton,
+		CloseIcon,
+		NcRichContenteditable,
+		NcNoteCard,
+		CreationIcon,
+		ClipboardCheckOutlineIcon,
+		ContentCopyIcon,
+	},
+	props: {
+		output: {
+			type: String,
+			default: '',
+		},
+		taskType: {
+			type: [Number, null],
+			default: null,
+		},
+	},
+	emits: [
+		'cancel',
+	],
+	data() {
+		return {
+			show: true,
+			myOutput: this.output,
+			closeButtonTitle: t('assistant', 'Close'),
+			closeButtonLabel: t('assistant', 'Close Nextcloud Assistant'),
+			modalSize: 'normal',
+			copied: false,
+		}
+	},
+	computed: {
+		taskTypeName() {
+			switch (this.taskType) {
+			case TASK_TYPES.text_generation:
+				return t('assistant', 'Text Generation')
+			case TASK_TYPES.speech_to_text:
+				return t('assistant', 'Speech to Text')
+			default:
+				return t('assistant', 'Unknown Result Type')
+			}
+		},
+	},
+	mounted() {
+		if (this.myOutput !== '') {
+			this.$refs.output.focus()
+		}
+	},
+	methods: {
+		onCancel() {
+			this.show = false
+			this.$emit('cancel')
+		},
+		async onCopy() {
+			try {
+				const container = this.$refs.output.$el
+				await this.$copyText(this.output.trim(), container)
+				this.copied = true
+				setTimeout(() => {
+					this.copied = false
+				}, 5000)
+			} catch (error) {
+				console.error(error)
+				showError(t('assistant', 'Result could not be copied to clipboard'))
+			}
+		},
+		onReset() {
+			this.myOutput = this.output
+		},
+
+		delayedReset() {
+			// This is a hack to ensure the text box is updated
+			// when we reset the text since removing newlines or spaces
+			// from the end of the text does not trigger an update.
+
+			// Delete any trailing newlines
+			this.myOutput = this.myOutput.replace(/\n+$/, '')
+			this.myOutput += '.'
+
+			// Let the ui refresh before resetting the text
+			setTimeout(() => {
+				this.onReset()
+			}, 0)
+		},
+	},
+}
+</script>
+
+<style lang="scss">
+// this is to avoid scroll on the container and leave it to the result block
+.assistant-modal .modal-container {
+	display: flex !important;
+
+	//&__content {
+	//padding: 16px;
+	//}
+}
+</style>
+
+<style lang="scss" scoped>
+.assistant-modal--wrapper {
+	//width: 100%;
+	padding: 16px;
+	overflow-y: auto;
+}
+
+.assistant-modal--content {
+	width: 100%;
+	display: flex;
+	flex-direction: column;
+	align-items: center;
+	justify-content: center;
+	overflow-y: auto;
+
+	> h2 {
+		display: flex;
+		margin: 12px 0 20px 0;
+		.icon {
+			margin-right: 8px;
+		}
+	}
+
+	.editable-output {
+		width: 100%;
+		display: flex;
+		flex-direction: column;
+		align-items: start;
```

Comment:
```suggestion
```
This is causing NcRichContenteditable to shrink to the width of the text content.